### PR TITLE
Introduce hoogle support

### PR DESCRIPTION
**Pull request by charleso**
/cc @jystic

### CODE REVIEW

File: main/mafia.hs [hunk 1]
```diff
@@ -613,3 +625,68 @@ sandbox_ :: Argument -> [Argument] -> EitherT MafiaViolation IO ()
...
+
+hoogle :: Text -> [Argument] -> EitherT MafiaViolation IO ()
+hoogle hackageRoot args = do
+  initialize
+  db <- ensureMafiaDir "hoogle"
+  hoogleExe <- installBinary "hoogle" "4.2.43" [("happy", "1.19.5")]
+  Out pkgStr <- sandbox "hc-pkg" ["list"]
+  let pkgs = fmap T.strip . filter (T.isPrefixOf " ") . T.lines $ pkgStr
+  hoos <- for pkgs $ \pkg -> do
+    let name = T.intercalate "-" . List.init $ T.splitOn "-" pkg
```
Comment (jacobstanley) [inline main/mafia.hs:651]: Why do we take the `init` just out of interest?

File: main/mafia.hs [hunk 2]
```diff
@@ -613,3 +625,68 @@ sandbox_ :: Argument -> [Argument] -> EitherT MafiaViolation IO ()
...
+    let txt = db </> pkg <> ".txt"
+    let hoo = db </> pkg <> ".hoo"
+    unlessM (doesFileExist txt) $ do
```
Comment (jacobstanley) [inline main/mafia.hs]: nice, i didn't know about `unlessM`

Comment (domdere) [inline main/mafia.hs]: It's an ambiata `P` special

**Comment by jacobstanley** [comment]
Amazing :100:

File: mafia.cabal [hunk 3]
```diff
@@ -19,11 +19,13 @@ library
                     , async                           == 2.0.*
                     , bytestring                      == 0.10.*
                     , containers                      == 0.5.*
+                    , cryptohash                      == 0.11.*
```
Comment (nhibberd) [inline mafia.cabal:22]: Probably don't need this on the library

File: mafia.cabal [hunk 4]
```diff
@@ -19,11 +19,13 @@ library
...
                     , directory                       == 1.2.*
                     , exceptions                      == 0.8.*
                     , filepath                        == 1.4.*
                     , process                         == 1.2.*
                     , tar                             == 0.4.*
+                    , temporary                       == 1.2.*
```
Comment (nhibberd) [inline mafia.cabal:28]: Same here maybe

**Comment by nhibberd** [comment]
Tests would be good :smiling_imp:

**Comment by nhibberd** [comment]
Looks good :100:

**Comment by charleso** [comment]
Added a sanity test. Let me know if anyone has problems using this.

**Pull request merged**